crates/router/src/core/cache.rs

```diff
@@ -0,0 +1,21 @@
+use common_utils::errors::CustomResult;
+
+use super::errors;
+use crate::{
+    cache::{ACCOUNTS_CACHE, CONFIG_CACHE},
+    db::StorageInterface,
+    services,
+};
+
+pub async fn invalidate(
+    store: &dyn StorageInterface,
+    key: &str,
+) -> CustomResult<services::api::ApplicationResponse<serde_json::Value>, errors::ApiErrorResponse> {
+    CONFIG_CACHE.remove(key).await;
+    ACCOUNTS_CACHE.remove(key).await;
+
+    match store.get_redis_conn().delete_key(key).await {
+        Ok(_del_reply) => Ok(services::api::ApplicationResponse::StatusOk),
+        _ => Err(errors::ApiErrorResponse::InvalidRequestUrl.into()),
```

Comment:
Here, we can match on `del_reply` and return a 200, 404 or 500 status code, something like this:

```rust
Ok(DelReply::KeyDeleted) => Ok(services::api::ApplicationResponse::StatusOk),
Ok(DelReply::KeyNotDeleted) => Err(errors::ApiErrorResponse::InvalidRequestUrl.into()),
error => error.into_report().change_context(errors::ApiErrorResponse::InternalServerError).attach_printable("Failed to invalidate cache")
```

You'd have to run the formatter by the way.

Reply:
done